tempest/api/workloadmgr/barbican/test_barbican.py

```diff
@@ -2833,3 +2833,142 @@ def test_8_barbican(self):
+    def test_9_barbican(self):
+        try:
+            test_var = "tempest.api.workloadmgr.barbican.test_"
+            tests = [[test_var + "workload_edit_with_unencrypted_and_encrypted_volume", 0]]
+            reporting.add_test_script(tests[0][0])
+
+            #create key pair...
+            kp = self.create_key_pair(tvaultconf.key_pair_name)
+
+            #create vm...
+            vm_id = self.create_vm(key_pair=kp)
+
+
+            #find unencrypted volume type. Get the volume_type_id
+            vol_type_id = -1
+            for vol in CONF.volume.volume_types:
+                if (vol.lower().find("luks") == -1):
+                    vol_type_id = CONF.volume.volume_types[vol]
+
+            if (vol_type_id == -1):
+                reporting.add_test_step("No unencrypted volume available to attach", tvaultconf.FAIL)
+                reporting.set_test_script_status(tvaultconf.FAIL)
+                raise Exception("No unencrypted volume available to attach...")
+
+            #Now create volume with derived volume type id...
+            volume_id = self.create_volume(
+                volume_type_id=vol_type_id)
+
+            #Attach volume to vm...
+            self.attach_volume(volume_id, vm_id)
+
+            #create a workload with encryption status as disabled and try to attach it to vm...
+            try:
+                wid = self.workload_create([vm_id],
+                                                tvaultconf.workload_type_id, encryption=False,
+                                                workload_cleanup=True)
+
+            except Exception as e:
+                LOG.error("Exception: " + str(e))
+                reporting.add_test_step("Unencrypted Workload creation with unencrypted volume is failed to create.", tvaultconf.FAIL)
+                reporting.set_test_script_status(tvaultconf.FAIL)
+
+            #create snapshot of workload created...
+            snapshot_id = self.workload_snapshot(wid, True)
+
+            self.wait_for_workload_tobe_available(wid)
+            snapshot_status = self.getSnapshotStatus(wid, snapshot_id)
+
+            if(snapshot_status == "available"):
+                LOG.debug("Full snapshot created...")
+            else:
+                reporting.add_test_step("Full snapshot is not created...", tvaultconf.FAIL)
+                reporting.set_test_script_status(tvaultconf.FAIL)
+                raise Exception("Create full snapshot failed")
+
+            #now create another vm with encrypted volume...
+            #create another key pair...
+            kp1 = self.create_key_pair(tvaultconf.key_pair_name)
+
+            #create another vm...
+            vm_id1 = self.create_vm(key_pair=kp1)
+
+            #find volume_type = luks. So that existing encrypted volume type can be used.
+            #Get the volume_type_id
+            vol_type_id = -1
+            for vol in CONF.volume.volume_types:
+                if (vol.lower().find("luks") != -1):
```

Comment:
Fetch both encrypted and unencrypted volume types at one time itself. If any of the volume type is not available, workload and snapshot steps can be skipped.